action.yml

```diff
@@ -69,30 +69,36 @@ runs:
     - name: Decrypt cached session file
       id: decrypt
       if: ${{ steps.restore-cache.outcome == 'success' }}
-      continue-on-error: true
       shell: bash
       run: |
+        # Default to no session.
+        echo "session_found=false" >> $GITHUB_OUTPUT
 
         # Verify that the encrypted session file was restored from cache.
-        test -s ${{ steps.configure-cache.outputs.path }}
+        if [ ! -s "${{ steps.configure-cache.outputs.path }}" ]; then
+          echo "No session file found in cache."
+          exit 0
+        fi
 
         # Decrypt the session file.
         echo "${{ inputs.pantheon-machine-token }}" | \
         openssl enc -d -aes-256-cbc -pbkdf2 -iter 10000 -pass stdin -in ${{ steps.configure-cache.outputs.path }} -out $HOME/.terminus/cache/session
 
-        # Check if restored session is still valid
-        TERMINUS_USER=$(terminus auth:whoami)
+        # Check for restored user (string) or allow command to fail gracefully.
+        TERMINUS_USER=$(terminus auth:whoami || true)
 
+        # Check TERMINUS_USER for string.
         if [ -z "$TERMINUS_USER" ]; then
           echo "No valid session found. "
-          exit 1
+          exit 0
         fi
```

Comment:
```
        # Check for restored user (string) or allow command to fail gracefully.
        TERMINUS_USER=""
        if ! TERMINUS_USER=$(terminus auth:whoami); then
          echo "terminus auth:whoami failed."
          echo "No valid session found."
          exit 0
        fi

        # Also assert TERMINUS_USER is a string.
        if [ -z "$TERMINUS_USER" ]; then
          echo "No valid session found. "
          exit 0
        fi
```
the `or true` inside the subshell smells to me, although I can't actually rationalize to myself why it's _not_ fine. Setting TERMINUS_USER _does_ prevent the assignment from masking the exit code.